### ISSUE TITLE
Image preview aspect ratio incorrect

### ISSUE THREAD

**Issue by ZhangTianrong** [sxyazi/yazi#2074]
### What system are you running Yazi on?

Windows

### What terminal are you running Yazi in?

Windows Terminal v1.22.3232.0

### `yazi --debug` output

```Shell
Yazi
    Version: 0.4.2 (5cfcab3 2024-12-20)
    Debug  : false
    Triple : x86_64-pc-windows-msvc (windows-x86_64)
    Rustc  : 1.83.0 (90b35a62 2024-11-26)

Ya
    Version: 0.4.2 (5cfcab3 2024-12-20)

Emulator
    Brand.from_env      : Some(Microsoft)
    Emulator.detect     : Emulator { kind: Left(Microsoft), light: false, cell_size: Some((10, 20)) }
    Emulator.detect_full: Ok(Emulator { kind: Right(Unknown { kgp: false, sixel: true }), light: false, cell_size: Some((10, 20)) })

Adapter
    Adapter.matches: Sixel

Desktop
    XDG_SESSION_TYPE           : None
    WAYLAND_DISPLAY            : None
    DISPLAY                    : Some("127.0.0.1:0.0")
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

NVIM
    NVIM          : false
    Neovim version: 0.10.2

Variables
    SHELL           : None
    EDITOR          : Some("vim")
    VISUAL          : None
    YAZI_FILE_ONE   : Some("C:\\Users\\<username>\\Scoop\\apps\\git\\current\\usr\\bin\\file.exe")
    YAZI_CONFIG_HOME: Some("C:\\Users\\<username>\\.config\\yazi")
    YAZI_ZOXIDE_OPTS: None
    FZF_DEFAULT_OPTS: None

Text Opener
    default     : Some(Opener { run: "code %*", block: false, orphan: true, desc: "code", for_: None, spread: true })
    block-create: Some(Opener { run: "code -w %*", block: true, orphan: false, desc: "code (block)", for_: None, spread: true })
    block-rename: Some(Opener { run: "code -w %*", block: true, orphan: false, desc: "code (block)", for_: None, spread: true })

Multiplexers
    TMUX               : 0
    tmux version       : program not found
    tmux build flags   : enable-sixel=Unknown
    ZELLIJ_SESSION_NAME: None
    Zellij version     : program not found

Dependencies
    file          : 5.45
    ueberzugpp    : program not found
    ffmpeg/ffprobe: 7.1 / 7.1
    pdftoppm      : 24.08.0
    magick        : 7.1.1-41
    fzf           : 0.57.0
    fd/fdfind     : 10.2.0 / program not found
    rg            : 14.1.1
    chafa         : 1.14.5
    zoxide        : 0.9.6
    7z/7zz        : 24.09 / program not found
    jq            : 1.7.1

Clipboard
    wl-copy/paste: program not found / program not found
    xclip        : program not found
    xsel         : program not found
```

### Describe the bug

The preview is always using a wrong aspect ratio. For instance, I have created a sample image of a 100x100 white square with
```
magick.exe convert -size 100x100 xc:white image.png
```
but it is rendered as a 160x205 white square in the preview.
![PixPin_2024-12-20_15-21-58](https://github.com/user-attachments/assets/59899fbd-2347-4f68-b1ae-4fec4722a84c)

### Minimal reproducer

Set the terminal font to be one with powerline/nerd and CJK support like [Sarasa Term SC Nerd Font](https://github.com/laishulu/Sarasa-Term-SC-Nerd) or [Consolas with Yahei](https://github.com/crvdgc/Consolas-with-Yahei). Create a sample image as mentioned above using `magick.exe convert -size 100x100 xc:white image.png` and then use `yazi` with the default config to preview the image.

### Anything else?

Given that `chafa` results in the same aspect ratio, it might be something related to the pixel height and width. Can we manually set it to something else like with `chafa`'s `--font-ratio` argument.

### Checklist

- [X] I tried the [latest nightly build](https://yazi-rs.github.io/docs/installation#official-binaries), and the issue is still reproducible
- [X] I updated the debug information (`yazi --debug`) input box to the nightly that I tried
- [X] I can reproduce it after disabling all custom configs/plugins (`mv ~/.config/yazi ~/.config/yazi-backup`)

**Comment by sxyazi** [comment]
I can't reproduce it by following your steps:

![screenshot-002201](https://github.com/user-attachments/assets/e9859ca9-46f6-41c2-93ba-6a5c9b5fbef1)

Are you sure you're using the default config right? Can you reproduce it on the latest nightly 0.4.3?

**Comment by ZhangTianrong** [comment]
> I can't reproduce it by following your steps:

I guess it's due to the different fonts we use. Please try setting the terminal to use [Sarasa-Term-SC-Nerd](https://github.com/laishulu/Sarasa-Term-SC-Nerd) to see if you can reproduce the problem. For me, the issue persists if I use another powerline font with CJK support like [Consolas-with-Yahei](https://github.com/laishulu/Sarasa-Term-SC-Nerd), but goes away if I switch to the original Consolas font. That is why I was specifically bringing up the `--font-ratio` option in `chafa` as I think the issue is likely related to how the fonts are created/patched.

**Comment by sxyazi** [comment]
OK, after applying the Sarasa-Term-SC-Nerd font to the MS Terminal, I can reproduce the issue.

However, I can't reproduce it in any of the other terminals I have installed that support the Sixel graphics format (iTerm2, WezTerm). They all use the Sarasa-Term-SC-Nerd font, and the graphics maintain the correct aspect ratio:

![screenshot-002202](https://github.com/user-attachments/assets/e829367d-f614-4487-9d71-8f51c369d7eb)

So, I suspect this is a bug with MS Terminal - it should maintain the image's aspect ratio when rendering graphics, since it's an image, not text. This isn't too surprising because they recently added Sixel support and it's still in the preview phase.

In this case, I don't think adding `--font-ratio` is reasonable, since it only addresses an MS Terminal-specific bug, and it doesn't seem to be very user-friendly - users should always preview images with the correct aspect ratio, no matter what font they're using, without having to manually adjust any configurations.

I suggest you report this issue to MS Terminal to request their support. Closing as not Yazi
